services/ui-src/src/components/drawer/Drawer.tsx

```diff
@@ -0,0 +1,136 @@
+import React from "react";
+
+// Components
+import {
+  Button,
+  Drawer as ChakraDrawer,
+  DrawerBody,
+  DrawerFooter,
+  DrawerHeader,
+  DrawerOverlay,
+  DrawerContent,
+  Flex,
+  Text,
+} from "@chakra-ui/react";
+//import { Form } from "components";
+import { CloseIcon } from "@cmsgov/design-system";
+
+//import { makeMediaQueryClasses } from "utils";
+
+export const Drawer = ({ drawerState }: Props) => {
+  //   const mqClasses = makeMediaQueryClasses();
+  return (
+    <ChakraDrawer
+      isOpen={drawerState.isOpen}
+      onClose={drawerState.onClose}
+      size="full"
+      placement="right"
+    >
+      <DrawerOverlay />
+      <DrawerContent sx={sx.drawerContent}>
+        <Flex sx={sx.drawerCloseContainer}>
+          <Button
+            sx={sx.drawerClose}
+            leftIcon={<CloseIcon />}
+            variant="link"
+            onClick={drawerState.onClose}
+          >
+            Close
+          </Button>
+        </Flex>
+        <DrawerHeader sx={sx.drawerHeader}>
+          <Text sx={sx.drawerHeaderText}>
+            Add a quality & performance measure
+          </Text>
+        </DrawerHeader>
+        <DrawerBody sx={sx.drawerBody}>
+          {/* TODO: Get Form details somehow */}
+          {/* <Form /> */}
+          <form>Placeholder</form>
+        </DrawerBody>
+        <DrawerFooter sx={sx.drawerFooter}>
+          <Button sx={sx.action}>Save</Button>
+          <Button sx={sx.close} onClick={drawerState.onClose}>
+            Close
+          </Button>
+        </DrawerFooter>
```

Comment:
I've updated the modal component on the dashboard to do the following:
```
        <ModalBody>{children}</ModalBody>

        <ModalFooter>
          <Button
            onClick={!actionId ? () => actionFunction() : undefined}
            form={actionId}
            type="submit"
          >
            {content.actionButtonText}
          </Button>
          <Button
            variant="link"
            onClick={modalState.onClose}
          >
            {content.closeButtonText}
          </Button>
        </ModalFooter>
```

That lets you pass in a form as a child in the parent component like so:

```
      <Modal>
        <Form
          id={formJson.id}
          formJson={formJson}
          formSchema={formSchema}
          onSubmit={addProgram}
        />
      </Modal>
```

Now that code will work though it hasn't been reviewed. Might be helpful though?

Reply:
> 

That worked! Thanks, @ntsummers1 🥇 

<img width="1104" alt="Screen Shot 2022-08-17 at 9 32 43 AM" src="https://user-images.githubusercontent.com/99458559/185147261-8c6cef08-2634-40c8-b1c5-8a66bb97d324.png">